examples/diagram.dx

```diff
@@ -0,0 +1,193 @@
+def listLength ((AsList n _):List a) : Int = n
+
+-- TODO: we want this for any monoid but this implementation won't work.
+def concat (lists:n=>(List a)) : List a =
+  totalSize = sum for i. listLength lists.i
+  AsList _ $ fst $ withState 0 \listIdx.
+    fst $ withState 0 \eltIdx.
+      for i:(Fin totalSize).
+        while (\(). get eltIdx >= listLength (lists.((get listIdx)@_))) \().
+          eltIdx := 0
+          listIdx := get listIdx + 1
+        (AsList _ xs) = lists.((get listIdx)@_)
+        eltIdxVal = get eltIdx
+        eltIdx := eltIdxVal + 1
+        xs.(eltIdxVal@_)
+
+
+Point : Type = (Float & Float)
+
+data Geom =
+  Spot
+  Circle Float
+  Rectangle Float Float  -- width, height
+  Line Point
+
+-- HTML color string. TODO: consider an RGB triple instead
+Color : Type = String
+
+-- TODO: we should add overloaded string literals so we don't need this
+def str (n:Int) ?-> (s:(Fin n=>Char)) : String = AsList _ s
+
+black : Color = str "black"
+white : Color = str "white"
+gray  : Color = str "gray"
+red   : Color = str "red"
+green : Color = str "green"
+blue  : Color = str "blue"
+
+Attributes : Type = { fill   : Maybe Color
```

Comment:
This is a pretty generic name if this is going to be included in a lot of examples. Maybe `DiagramAttributes` would be clearer?